File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-10.rs
impl Solution {
    pub fn length_of_longest_substring(s: String) -> i32 {
        let mut l:usize=0;
        let mut r:usize=0;
        let s_vec:Vec<char>=s.chars().collect();
        let len:usize=s_vec.len();
        let mut max:i32=0;
        let mut set:HashSet<char>=HashSet::new();
        while r<len{
            while r<len && !set.contains(&s_vec[r]){
                set.insert(s_vec[r]);
                r+=1;
            }
            max=max.max(set.len() as i32);
            set.clear();
            l+=1;
            r=l;
        }
        return max;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-10.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int len=nums.length;
        Set<Integer> set=new HashSet<>();
        for(int num:nums){
            set.add(num);
        }
        int max=0;
        for(int num:nums){
            if(!set.contains(num-1)){
                int count=0;
                while(set.contains(num)){
                    num++;
                    count++;
                }
                max=Math.max(count, max);
            }
        }
        return max;
    }
}
